src/Microsoft.Health.Fhir.Core/Features/Operations/BulkDelete/BulkDeleteProcessingJob.cs

```diff
@@ -105,7 +122,20 @@ public async Task<string> ExecuteAsync(JobInfo jobInfo, CancellationToken cancel
                     }
                 }
 
-                await _mediator.Publish(new BulkDeleteMetricsNotification(jobInfo.Id, resourcesDeleted.Sum(resource => resource.Value)), cancellationToken);
+                try
+                {
+                    var notification = new BulkDeleteMetricsNotification(jobInfo.Id, resourcesDeleted.Count)
+                    {
+                        Content = CreateNotificationContent(deletedSearchParameters),
+                    };
+
+                    await _mediator.Publish(notification, cancellationToken);
```

Comment:
My only concern with this method of updating the search parameters is that it isn't safe if the service shuts down. If the notification isn't fully handled the DB can be left in a bad state where the Search Parameter resources are deleted, but the Search Parameter table hasn't been updated yet.